mne_bids/copyfiles.py

```diff
@@ -616,13 +617,28 @@ def copyfile_bti(raw, dest):
     copyfile_kit
 
     """
-    pdf_fname = "c,rfDC"
-    if raw.info["highpass"] is not None:
-        pdf_fname = "c,rf%0.1fHz" % raw.info["highpass"]
-    sh.copyfile(raw._init_kwargs["pdf_fname"], op.join(dest, pdf_fname))
-    sh.copyfile(raw._init_kwargs["config_fname"], op.join(dest, "config"))
-
-    # If no headshape file present, cannot copy it
-    hs_file = raw._init_kwargs.get("head_shape_fname")
-    if hs_file is not None:  # pragma: no cover
-        sh.copyfile(hs_file, op.join(dest, "hs_file"))
+    # XXX: remove after support for mne<1.5 is dropped
+    # BTi operations were unreliable in mne-bids prior to mne 1.5 / mne-bids 0.13
+    need_mne = "1.3"
+    if _compare_version(mne.__version__, "<", need_mne):
+        raise ImportError(
+            f"mne >= {need_mne} is required for BTi operations, got {mne.__version__}"
+        )
```

Comment:
@larsoner this `_compare_version` works in unexpected ways.

MNE's version is currently something like: `'1.4.0.dev224+g0fd4ded6b'`.

IMHO this should work (evaluate `True`): `_compare_version(mne.__version__, "<", 1.4):` ... because we're in the dev branch AFTER the 1.4 release. However, it doesn't evaluate to `True` (hence me setting it to 1.3 in the code).

Another question. When I make release in other softwares (including mne-bids), I:

- release a stable version (e.g., 0.12)
- then start developing the next version (e.g., 0.13)
- hence my dev version is called 0.13.dev0
- if I release 0.12.1 (and not 0.13) in the meantime, I want to switch back to a new dev version
- that would be called 0.13.dev1 ... and so on

do I get something wrong here? Where does mne's versioning scheme fit in here?

Reply:
IMHO the check is correct **but** the version number is not.

It should not be `1.4.0.devX` which AFAIK indicates a dev version prior to a 1.4.0 release:

> The developmental release segment consists of the string .dev, followed by a non-negative integer value. Developmental releases are ordered by their numerical component, immediately before the corresponding release (and before any pre-releases with the same release segment), and following any previous release (including any post-releases).
> 
> Developmental releases are also permitted for pre-releases and post-releases:
> 
> X.YaN.devM       # Developmental release of an alpha release
> X.YbN.devM       # Developmental release of a beta release
> X.YrcN.devM      # Developmental release of a release candidate
> X.Y.postN.devM   # Developmental release of a post-release
> 

https://peps.python.org/pep-0440/#developmental-releases

So the version number of our current dev version should be `1.4.0.post0.dev0`

Reply:
okay thanks @hoechenberger this is also my understanding

Reply:
> Or `1.5.0.dev0`, which I wouldn't know how to do with `setuptools-scm` without adding a new tag

Actually this should happen automatically:

https://github.com/pypa/setuptools_scm#default-versioning-scheme

> The next version is calculated by adding 1 to the last numeric component of the tag.


so we're screwing that up somehow.

Reply:
@sappelhoff I ran pip install again and now the version number seems correct…

Reply:
why? What changed? I am still getting `mne-1.4.0.dev237+ga6c07cb5c` after a fresh pull from upstream and an editable pip install

Reply:
May have to do with the fact that we're writing the version to `mne/_version.py` during install
And after that it doesn't get updated anymore

Reply:
what's the version you are getting?

Reply:
Now I'm getting 1.4 again … ummm … I have no idea what's going on

And in mne/_version.py it still says

```
__version__ = version = '1.5.0.dev88+ga6c07cb5c'
__version_tuple__ = version_tuple = (1, 5, 0, 'dev88', 'ga6c07cb5c')
```

Reply:
:thinking: 
```
# file generated by setuptools_scm
# don't change, don't track in version control
__version__ = version = '1.4.0.dev237+ga6c07cb5c'
__version_tuple__ = version_tuple = (1, 4, 0, 'dev237', 'ga6c07cb5c')
```

Reply:
Yikes something is wrong:
```
$ pip install --no-build-isolation -ve .
...
Successfully installed mne-1.5.0.dev75+g0fd4ded6b
$ python -c "import mne; print(mne.__version__)"
1.4.2
$ python -c "import mne; print(mne)"
<module 'mne' from '/Users/larsoner/python/mne-python/mne/__init__.py'>
$ cat mne/_version.py 
# coding: utf-8
# file generated by setuptools_scm
# don't change, don't track in version control
__version__ = version = '1.5.0.dev75+g0fd4ded6b'
__version_tuple__ = version_tuple = (1, 5, 0, 'dev75', 'g0fd4ded6b')
```
I wonder if this has been broken ever since we release 1.4.1 / 1.4.2. I'll look a bit...

Reply:
I believe it's the call to

```
    from importlib.metadata import version

    __version__ = version("mne")
```

in MNE's `__init__.py` that's the problem

Reply:
For me the presence of a `mne.egg-info` made it mess up the search when in the MNE-Python directory:
```
mne-python larsoner $ rm -Rf mne.egg-info
mne-python larsoner$ python -c "import mne; print(mne, mne.__version__)"
<module 'mne' from '/Users/larsoner/python/mne-python/mne/__init__.py'> 1.5.0.dev75+g0fd4ded6b
```
I figured this out by looking at and seeing something like this
```
>>> from importlib.metadata import distribution, MetadataPathFinder
>>> d = distribution("mne")
>>> d._path
PosixPath('mne.egg-info')
>>> list(x._path for x in MetadataPathFinder.find_distributions() if "mne" == x.name)
[PosixPath('mne.egg-info'), PosixPath('/Users/larsoner/opt/miniconda3/envs/mne/lib/python3.10/site-packages/mne-1.5.0.dev75+g0fd4ded6b.dist-info')]
```

Reply:
I can't reproduce these results, I keep getting `1.4.0.dev237+ga6c07cb5c`

Reply:
@sappelhoff my only idea is maybe you haven't fetched the MNE tags from `upstream` in a while. The version is built off the tags in the repo.

If this fails, too, then try `git clone`ing just upstream and doing an editable install. If this also fails, maybe your `setuptools` or `setuptools_scm` is too old. Have you tried upgrading these and doing `pip install --no-build-isolation -ve .` or so? Then it will use your locally installed versions of these deps rather than isolating them.

Reply:
I don't have this file on my Linux machine (the above offender was working on my macOS machine) and the CIs haven't had this problem, so I'm hoping it's something transient / leftover on our machines from a while ago. I even tried `python setup.py bdist_wheel sdist` locally on my Linux machine and couldn't reproduce the problem. So hopefully it's something we can ignore and won't be a problem going forward, but if we see it happen again and can figure out the root cause we should definitely fix it!

Reply:
... actually doing `python setup.py bdist_wheel sdist` *did* create the file on my Linux machine! So that's something to look into. I'll create a new issue in MNE about this.